extension/encoding/awslogsencodingextension/internal/unmarshaler/vpc-flow-log/vpc_flow_log_unmarshaler.go

```diff
@@ -0,0 +1,196 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package vpcflowlog // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/encoding/awslogsencodingextension/internal/unmarshaler/vpc-flow-log"
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/klauspost/compress/gzip"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/plog"
+	conventions "go.opentelemetry.io/collector/semconv/v1.27.0"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/encoding/awslogsencodingextension/internal/metadata"
+)
+
+const (
+	fileFormatPlainText = "plain-text"
+	fileFormatParquet   = "parquet"
+)
+
+var (
+	supportedVPCFlowLogFileFormat = []string{fileFormatPlainText, fileFormatParquet}
+
+	supportedFieldValues = map[string][]string{
```

Comment:
If/when we add support for ECS fields, we should follow https://opentelemetry.io/docs/specs/semconv/resource/cloud-provider/aws/ecs/

Reply:
Fields added. See https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/38897#discussion_r2012241861. Closing this conversation in favor of that.